src/ui/api/email.go

```diff
@@ -32,20 +30,20 @@ const (
 
 // EmailAPI ...
 type EmailAPI struct {
-	api.BaseAPI
+	BaseController
 }
 
 // Prepare ...
 func (e *EmailAPI) Prepare() {
-	userID := e.ValidateUser()
-	isSysAdmin, err := dao.IsAdminRole(userID)
-	if err != nil {
-		log.Errorf("failed to check the role of user: %v", err)
-		e.CustomAbort(http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError))
+	e.BaseController.Prepare()
```

Comment:
Is there any better way to have a general rule to check the authentication?  I see the duplicate code block in several APIs.